### PR TITLE
Initial Spring Boot / Hibernate App

### PR DESCRIPTION
**Pull request by jb-adams**
Creates an initial Java application for the testbed API
* Gradle for dependency management
* Spring Boot for server/REST API
* Hibernate for database object-relational mapping
* Database ERD has been replicated as Java classes with Hibernate API

@yash-puligundla there are some test controller functions that load entities from the test db and write out as JSON. Please try to install and run with:
```
make sqlite-db-refresh
./gradlew bootRun
```

Then try to hit the sample controller functions at:
```
GET http://localhost:4500/test/specification
GET http://localhost:4500/test/testbed
GET http://localhost:4500/test/platform
GET http://localhost:4500/test/organization
GET http://localhost:4500/test/report-series
GET http://localhost:4500/test/report
```

### CODE REVIEW

**Comment by yash-puligundla** [comment]
Hi @jb-adams, Great progress! Could you please clarify what is the expected behavior of the GET endpoints for this Pull Request? 
I am trying to understand if the _GET specification_ endpoint is expected to return a list of all the specifications present in the database or just the one that matches the provided id.

File: src/main/java/org/ga4gh/testbed/api/model/TestbedTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private Summary summary;
+
+    @OneToMany(mappedBy = "testbedTest",
+               fetch = FetchType.LAZY,
+               cascade = CascadeType.ALL,
+               orphanRemoval = true)
+    private List<TestbedCase> testbedCases;
+
+    @ManyToOne
+    @JoinColumn(name = "fk_phase_id")
```
[the remark this answers is not in the document]
Comment (jb-adams) [reply src/main/java/org/ga4gh/testbed/api/model/TestbedTest.java:74]: I don't know if we can add "nullable = false" for relational/foreign key columns. The reason being is that some relationships are lazily loaded, i.e. only loaded if explicitly asked for. In the `/test/report` example, I haven't loaded the `phases` related `tests`. This is useful when listing out objects (i.e. reports), and you don't want to query for all the sub objects of each report in the list.

But I have added nullable = false for non-foreign key attributes

File: src/main/java/org/ga4gh/testbed/api/model/TestbedTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+@Getter
+@NoArgsConstructor
+public class TestbedTest implements HibernateEntity<Integer> {
+
+    @Id
+    @Column(name = "id", updatable = false, nullable = false)
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+
+    @Column(name = "test_name")
```
[the remark this answers is not in the document]
Comment (jb-adams) [reply src/main/java/org/ga4gh/testbed/api/model/TestbedTest.java]: I've added nullable = false for all non-foreign key attributes, where NOT NULL is specified in the ERD

[the remark this answers is not in the document]
Comment (yash-puligundla) [reply src/main/java/org/ga4gh/testbed/api/model/TestbedTest.java]: This sounds good. Thanks, Jeremy!

**Comment by yash-puligundla** [comment]
I was not able to find the "role" field from the "github_user_organization" table. Could you please take a look? @jb-adams

**Comment by jb-adams** [comment]
> I was not able to find the "role" field from the "github_user_organization" table. Could you please take a look? @jb-adams

You are correct, it's not currently in there. It's a bit more complex because it's an extra column on the join table, which the current @ManyToMany annotation can't quite capture. There appears to be a tutorial [here](https://mkyong.com/hibernate/hibernate-many-to-many-example-join-table-extra-column-annotation/) that manages to effectively capture extra attributes in a join table. I'll try to get it working for our table structure.

**Comment by yash-puligundla** [comment]
> > I was not able to find the "role" field from the "github_user_organization" table. Could you please take a look? @jb-adams
> 
> You are correct, it's not currently in there. It's a bit more complex because it's an extra column on the join table, which the current @manytomany annotation can't quite capture. There appears to be a tutorial [here](https://mkyong.com/hibernate/hibernate-many-to-many-example-join-table-extra-column-annotation/) that manages to effectively capture extra attributes in a join table. I'll try to get it working for our table structure.

Thanks for the clarification. The PR looks great!

**Comment by jb-adams** [comment]
> Thanks for the clarification. The PR looks great! @jb-adams Do you think we should keep this effort separate or keep it a part of this PR?

Let me take the time now to figure this out, since we will need to make use of that attribute

**Comment by jb-adams** [comment]
@yash-puligundla my latest commit should now display the role(s). Can you please test out the following endpoints:

```
GET http://localhost:4500/test/organization
GET http://localhost:4500/test/github-user
```

and verify you can see the role associated with a user-organization  pair

**Comment by yash-puligundla** [comment]
Perfect! works as expected. Approving!

**Review by yash-puligundla — APPROVED** [review]

**Pull request merged**